ui/lib/config-ui/addon/components/messages/tab-page-header.hbs

```diff
@@ -0,0 +1,51 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+~}}
+
+<PageHeader as |p|>
+  {{#if @breadcrumbs}}
+    <p.top>
+      <Page::Breadcrumbs @breadcrumbs={{@breadcrumbs}} />
+    </p.top>
+  {{/if}}
+  <p.levelLeft>
+    <Hds::Text::Display @tag="h2" @size="500" class="has-top-margin-xl">
+      Custom messages
+    </Hds::Text::Display>
+  </p.levelLeft>
+</PageHeader>
+
+<div class="tabs-container box is-bottomless is-marginless is-paddingless">
+  <nav class="tabs" aria-label="custom-messages">
+    <ul>
+      <LinkTo
+        class={{if @authenticated "active"}}
```

Comment:
An issue I ran into: when transitioning to different tabs, the page query params are retained from the previous tab. This is the default behavior since controllers are singletons. I tried adding a `resetController` to reset the page to 1. I noticed that we have explicitly state what the query params are in order for the tabs to transition with the correct query params. Here's a good description of what I saw happening: https://guides.emberjs.com/v2.14.0/routing/query-params/

Reply:
You should be able to reset to null. Let's zoom through this as it gets a little confusing. Ex:
```
resetController(controller, isExiting) {
    if (isExiting) {
      controller.set('pageFilter', null);
      controller.set('page', null);
    }
  }
```